pages/docs/reference/coding-conventions.md

```diff
@@ -726,6 +727,195 @@ foo {
 
 </div>
 
+### Trailing commas
+
+Use trailing commas at declaration site, including multi-line enumerations such as argument and parameter lists, `when` entries, and components of destructuring declarations.
+For call site trailing commas are optional.
+
+Using trailing commas has several benefits:
+
+* It makes version-control diffs cleaner: all focus is on changed value.
+* It's easy to add or reorder elements: no need to add or delete the comma sign if you manipulate elements.
+* It simplifies code generation (for example, for object initializers). The last element can also have the comma sign.
+
+Kotlin supports trailing commas in the following cases:
+* [Lists and enumerations](#lists-and-enumerations)
+* [Class parameters](#class-parameters)
+* [Function value parameters](#function-value-parameters)
+* [Parameters with optional type (including setters)](#parameters-with-optional-type-including-setters)
+* [Value arguments](#value-arguments)
+* [Indexing suffix](#indexing-suffix)
+* [Lambda parameters](#lambda-parameters)
+* [`when` entry](#when-entry)
+* [Collection literals (in annotations)](#collection-literals-in-annotations)
+
+> Kotlin doesn't support:
+> * Single trailing comma without elements (including declaration site).
+> * More than one trailing comma in a row.
+> * Trailing commas in the following cases:
+>    * Multiple variable declaration
+>    * Type parameters
+>    * Function type parameters
+>    * Type arguments
+>    * Delegation specifiers
+>    * Type constraints
+{:.note}
+
+#### Lists and enumerations
+
+<div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+```kotlin
+val colors = listOf(
+    "red",
+    "green",
+    "blue", // trailing comma
+)
+
+enum class Classes { 
+    Foo, 
+    Bar, // trailing comma
+}
+```
+
+</div>
+
+#### Class parameters
+
+<div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+```kotlin
+class A(
+    val x: String,
+    val y: String, // trailing comma
+)
+```
+
+</div>
+
+#### Function value parameters
+
+<div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+```kotlin
+fun foo(
+    x: Int,
+    y: Number, // trailing comma
+) {}
+
+constructor(
+    x: Comparable<Comparable<Number>>,
```

Comment:
Are you sure we need all these generics in generics for an example of trailing comma?